plugins/woocommerce/tests/e2e-pw/tests/merchant/page-loads.spec.js

```diff
@@ -104,6 +232,10 @@ for ( const currentPage of wcPages ) {
 				await expect(
 					page.locator( 'h1.components-text' )
 				).toContainText( currentPage.subpages[ i ].heading );
+
+				await expect(
+					page.locator( currentPage.subpages[ i ].element )
+				).toContainText( currentPage.subpages[ i ].text );
```

Comment:
I think we need to add a few more stricter assertions because the `toContainText` assertion will pass even though the element is not visible, as long as it's present in the DOM.

For example, I did an experiment with the `Can load Orders` test. I deleted these lines from the `beforeAll` hook:
```js
			// create an order
			await api
				.post( 'orders', {
					line_items: [
						{
							product_id: productId,
							quantity: 1,
						},
					],
				} )
				.then( ( response ) => {
					orderId = response.data.id;
				} );

```

Then I ran the `page-loads.spec.js` spec. Notice in the Playwright UI below that the `Can load Orders` test passed even though there were no orders appearing on the page.

<img width="1278" alt="Screenshot 2023-09-14 at 1 02 17 AM" src="https://github.com/woocommerce/woocommerce/assets/4509348/1a6a2b0a-ec02-482b-8dfa-de2416fc5b88">

This is because the element `.select2-selection__placeholder` is present in the DOM but not visible as shown below:

<img width="1428" alt="Screenshot 2023-09-14 at 12 58 40 AM" src="https://github.com/woocommerce/woocommerce/assets/4509348/c4d9fd95-b9b0-433e-806c-4a75e01cbcbe">

So if there's a bug where the `Filter by registered customer` menu isn't showing up, then the `Can load Orders` test would still pass, resulting to a dangerous false negative. 

Would you consider adding stricter assertions aside from `toContainText()`, like `toBeVisible()` or some others?

Reply:
I kinda expected for `toContainText()` to look only for visible items. Good catch! Updated now with waiting for the element `toBeVisible()` and then verifying the `toContainText()`.